src/plugins/syscalls/syscalls.cpp

```diff
@@ -128,10 +130,99 @@ static event_response_t cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
     return 0;
 }
 
+static event_response_t win_cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
+    int i,j;
+
+    syscall_wrapper_t *wrapper = (syscall_wrapper_t*)info->trap->data;
+    syscalls *s = wrapper->sc;
+
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+
+    uint8_t reg_size = vmi_get_address_width(vmi);
+    int nargs = win_syscall_struct[wrapper->syscall_index].num_args;
+
+    access_context_t ctx;
+    ctx.translate_mechanism = VMI_TM_PROCESS_DTB;
+    ctx.dtb = info->regs->cr3;
+
+    int size = reg_size * nargs; 
+    unsigned char buf[size];
+
+    if(reg_size==4){ // 32 bit os
+    
+      ctx.addr = info->regs->rsp+reg_size;  // jump over base pointer
+
+      // multiply num args by 4 for 32 bit systems to get the number of bytes we need
+      // to read from the stack.  assumes standard calling convention (cdecl) for the 
+      // visual studio compile.
+      if(size != vmi_read(vmi, &ctx, buf, size)){
+        return 0;
+      }
+    }
+    else { // 64 bit os - ************** UNTESTED *******************!
+      ctx.addr = info->regs->rcx;
+      if(VMI_FAILURE == vmi_read_64(vmi, &ctx, (uint64_t *)&buf)){
+          return 0;
+      }
+      ctx.addr = info->regs->rdx;
+      if(VMI_FAILURE == vmi_read_64(vmi, &ctx, (uint64_t *)&buf[reg_size])){
+          return 0;
+      }
+      ctx.addr = info->regs->r8;
+      if(VMI_FAILURE == vmi_read_64(vmi, &ctx, (uint64_t *)&buf[reg_size*2])){
```

Comment:
There is no need to read anything here, the argument value is in the register itself.